Mathlib/RingTheory/Bialgebra/TensorProduct.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2024 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.RingTheory.Bialgebra.Equiv
+import Mathlib.RingTheory.Coalgebra.TensorProduct
+import Mathlib.RingTheory.TensorProduct.Basic
+
+/-!
+# Tensor products of bialgebras
+
+We define the data in the monoidal structure on the category of bialgebras - e.g. the bialgebra
+instance on a tensor product of bialgebras, and the tensor product of two `BialgHom`s as a
+`BialgHom`. This is done by combining the corresponding API for coalgebras and algebras.
+
+## Implementation notes
+
+Since the coalgebra instance on a tensor product of coalgebras is defined using a universe
+monomorphic monoidal category structure on `ModuleCat R`, we require our bialgebras to be in the
+same universe as the base ring `R` here too. Any contribution that achieves universe polymorphism
+would be welcome. For instance, the tensor product of bialgebras in the
+[FLT repo](https://github.com/eric-wieser/FLT/blob/8764e28f03cd0d03f673e1288f47fc7a665207de/FLT/for_mathlib/Coalgebra/TensorProduct.lean)
+is already universe polymorphic since it does not go via category theory.
+
+-/
+
+universe v u
+
+open scoped TensorProduct
+
+namespace Bialgebra.TensorProduct
+variable {R A B : Type u} [CommRing R] [Ring A] [Ring B]
+    [Bialgebra R A] [Bialgebra R B]
+
+lemma coe_counit_eq :
+    ⇑(Coalgebra.counit (R := R) (A := A ⊗[R] B))
+      = (Algebra.TensorProduct.lmul' R).comp (Algebra.TensorProduct.map
+      (counitAlgHom R A) (counitAlgHom R B)) := by
+  rfl
+
+lemma coe_comul_eq :
+    ⇑(Coalgebra.comul (R := R) (A := A ⊗[R] B))
+      = (Algebra.TensorProduct.tensorTensorTensorComm R A A B B).toAlgHom.comp
+      (Algebra.TensorProduct.map (comulAlgHom R A) (comulAlgHom R B)) := by
+  rfl
+
+end Bialgebra.TensorProduct
+
+noncomputable instance TensorProduct.instBialgebra
+    {R A B : Type u} [CommRing R] [Ring A] [Ring B]
+    [Bialgebra R A] [Bialgebra R B] : Bialgebra R (A ⊗[R] B) := by
+  refine Bialgebra.mk' R (A ⊗[R] B) ?_ (fun {x y} => ?_) ?_ (fun {x y} => ?_)
+  <;> simp only [Bialgebra.TensorProduct.coe_counit_eq, Bialgebra.TensorProduct.coe_comul_eq]
```

Comment:
I believe we are supposed to write `<;>` before the line break, as it is a kind of infix operator.